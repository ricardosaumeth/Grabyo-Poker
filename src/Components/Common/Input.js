import React, { Component } from 'react';

export default class Input extends Component {
  constructor(props) {
    super(props);

    this.state = {
      value: ''
    };

    this.textInput = React.createRef();
    this.onChangedInput = this.onChangedInput.bind(this);
  }

  onChangedInput(e) {
    this.setState({ value: e.target.value });
  }

  render() {
    const { playerName, isDisabled } = this.props;
    const { value } = this.state;
    return (
      <input
        type="text"
        placeholder={playerName}
        value={value}
        ref={this.textInput}
        onChange={this.onChangedInput}
        disabled={isDisabled}
      />
    );
  }
}
