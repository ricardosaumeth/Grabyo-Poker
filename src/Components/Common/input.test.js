import React from "react";
import "../../setupTests"

import Input from "./Input";
import renderer from 'react-test-renderer';
import { mount, shallow } from "enzyme";

describe(`Card deck`, () => {
	test('should match the Snapshot', () => {
		const wrapper = renderer.create(<Input  playerName='Player 1' isDisabled={true} />).toJSON();
		expect(wrapper).toMatchSnapshot();
	});
	test('should set state.value', () => {
    const wrapper = shallow(<Input  playerName='Player 1' isDisabled={true} />);
    wrapper.setState({ value: '1' });
		expect(wrapper.state().value).toEqual('1');
	});
});