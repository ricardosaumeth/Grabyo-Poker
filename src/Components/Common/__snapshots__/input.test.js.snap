// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card deck should match the Snapshot 1`] = `
<input
  disabled={true}
  onChange={[Function]}
  placeholder="Player 1"
  type="text"
  value=""
/>
`;
